terraform/modules/example/main.tf

```diff
@@ -36,30 +36,46 @@ data "template_file" "main" {
   }
 }
 
-# The autoscaling group
-module "asg" {
-  source  = "telia-oss/asg/aws"
-  version = "0.2.0"
-
-  name_prefix       = "${var.name_prefix}"
-  user_data         = "${data.template_file.main.rendered}"
-  vpc_id            = "${var.vpc_id}"
-  subnet_ids        = "${var.subnet_ids}"
-  await_signal      = "true"
-  pause_time        = "PT5M"
-  health_check_type = "EC2"
-  instance_policy   = "${data.aws_iam_policy_document.permissions.json}"
-  min_size          = "${var.instance_count}"
-  instance_type     = "${var.instance_type}"
-  instance_ami      = "${var.instance_ami}"
-  instance_key      = "${var.instance_key}"
-  tags              = "${var.tags}"
+resource "aws_launch_configuration" "main" {
+  name_prefix            = "${var.name_prefix}"
+  image_id               = "${var.instance_ami}"
+  instance_type          = "${var.instance_type}"
+  key_name               = "${var.instance_key}"
+  iam_instance_profile   = "${aws_iam_instance_profile.lifecycle.name}"
+  security_groups        = ["${aws_security_group.main.id}"]
+
+  user_data = "${data.template_file.main.rendered}"
+
+  lifecycle {
+    create_before_destroy = true
+  }
+}
+
+resource "aws_autoscaling_group" "main" {
+  name                 = "${var.name_prefix}-${aws_launch_configuration.main.name}"
+  launch_configuration = "${aws_launch_configuration.main.id}"
+  vpc_zone_identifier  = ["${var.subnet_ids}"]
+
+  min_size         = "0"
+  desired_capacity = "${var.instance_count}"
+  max_size         = "1"
```

Comment:
Think this has to match `${var.instance_count}` (or we can just remove the `instance_count` variable all together).